soc/arm/atmel_sam/sam3x/soc.c

```diff
@@ -192,15 +193,7 @@ static ALWAYS_INLINE void clock_init(void)
 	}
 }
 
-/**
- * @brief Perform basic hardware initialization at boot.
- *
- * This has to be run at the very beginning thus the init priority is set at
- * 0 (zero).
- *
- * @return 0
- */
-static int atmel_sam3x_init(void)
+void z_arm_platform_init(void)
```

Comment:
`z_arm_platform_init()` is a lot earlier than `PRE_KERNEL_1` - are you confident that this will work in all cases?

Reply:
The z_arm_platform_init() seems to be the correct place to configure clocks, cache and any other specific details. This happen at reset which configure SoC ASAP. This speed up core initialization too.

The remaining PRE_KERNEL are initialization that require some kind OS resource.